components/UpdateProfile/UpdateProfile.jsx

```diff
@@ -47,83 +54,89 @@ export default function UpdateProfile({
   }
 
   return (
-    <div
-      className="stack inline-max"
-      style={{ "--max-inline-size": "var(--input-screen-inline-max-size" }}
-    >
-      <h1>Update profile</h1>
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogTrigger className="button" data-variant="primary">
+        Update Profile
+      </DialogTrigger>
 
-      <h2>Update avatar</h2>
-      <Avatar url={avatarUrl} size={250} />
-      <AddImage
-        uid={profileData.id}
-        setPublicUrl={(url) => setAvatarUrl(url)}
-      />
-      <br />
+      <DialogContent>
+        <header>
+          <h2>Update profile</h2>
+        </header>
 
-      <label className="label" htmlFor="username">
-        Username
-      </label>
-      <br />
-      <small>
-        This will change your profile URL. https://downloadcodemanager.com/
-        {`${sluggedName}`}
-      </small>
-      <input
-        className="input"
-        id="username"
-        type="text"
-        value={username || ""}
-        onChange={(e) => setUsername(e.target.value)}
-        required
-      />
-
-      <label className="label" htmlFor="location">
-        Location
-      </label>
-      <input
-        className="input"
-        id="location"
-        type="text"
-        value={location || ""}
-        onChange={(e) => setLocation(e.target.value)}
-      />
+        <div className="stack block-overflow">
+          <Avatar url={avatarUrl} size={250} />
+          <AddImage
+            uid={profileData.id}
+            setPublicUrl={(url) => setAvatarUrl(url)}
+          />
+          <br />
 
-      <input
-        id="passwordProtect"
-        type="checkbox"
-        checked={isPasswordProtected}
-        onChange={() => setIsPasswordProtected(!isPasswordProtected)}
-      />
-      <label className="label" htmlFor="passwordProtect">
-        Password protect page?
-      </label>
+          <label className="label" htmlFor="username">
+            Username
+          </label>
+          <br />
+          <small>
+            This will change your profile URL. https://downloadcodemanager.com/
```

Comment:
I recommend using a `process.env` value instead of setting the domain as a static string. [Netlify build environment variables](https://docs.netlify.com/configure-builds/environment-variables/#deploy-urls-and-metadata) include a  `process.env.URL` that contains the domain value. Locally you could set this in your `.env` file

Reply:
done done d